packages/js-lib/src/zod/zod.shared.schemas.test.ts

```diff
@@ -0,0 +1,57 @@
+import { getAjv } from '@naturalcycles/nodejs-lib'
+import { describe, expect, test } from 'vitest'
+import { customZodSchemas, z } from './index.js'
+
+test.each(Object.keys(customZodSchemas))(
+  'custom zod schemas like "z.%s" should properly convert to JSON schema and AJV schema',
+  key => {
+    const validator = customZodSchemas[key as keyof typeof customZodSchemas]
+    const zodSchema = z.object({ value: validator })
+    const jsonSchema = z.toJSONSchema(zodSchema, { target: 'draft-7' })
+    const ajvSchema = getAjv().compile(jsonSchema)
+
+    expect(ajvSchema).toBeDefined()
+  },
+)
+
+describe('z.email', () => {
+  test('should accept a valid email address', () => {
+    const email = 'test@example.com'
+    const result = z.email.parse(email)
+    expect(result).toBe(email)
+  })
+
+  test('should reject an email with a capital letter', () => {
+    const email = 'Test@example.com'
+    const result = z.email.safeParse(email)
+    expect(result.success).toBe(false)
+  })
+
+  test('should not trim before validation', () => {
+    const email = ' test@example.com '
+    const result = z.email.safeParse(email)
+    expect(result.success).toBe(false)
+  })
```

Comment:
Even if we add `z.string().trim().email()` as you originally did, 
the trimming will be removed from the JSON Schema,
and Ajv - by default - will not apply trimming.

I think it would be confusing to have `trim()` in the zod schema,
which is then stripped,
and we re-introduce it in Ajv by some default config.

E.g. someone tries to create a zod schema without trimming, only to find that the validation will still trim everything, due to Ajv default config.